CustomDimensions.php

```diff
@@ -280,10 +281,13 @@ public function addConversionInformation(&$conversion, $visitInformation, Tracke
 
         foreach ($dimensions as $dimension) {
             $index = (int) $dimension['index'];
-            if ($dimension['scope'] === self::SCOPE_VISIT && in_array($index, $conversionIndexes)) {
+            if (in_array($dimension['scope'], array(self::SCOPE_ACTION, self::SCOPE_VISIT)) && in_array($index, $conversionIndexes)) {
                 $field = LogTable::buildCustomDimensionColumnName($dimension);
 
-                if (array_key_exists($field, $visitInformation)) {
+                if ($dimension['scope'] === self::SCOPE_ACTION && array_key_exists($field, $actionDimensions)) {
```

Comment:
I haven't tested it but looks good now.  Maybe in ` else if (array_key_exists($field, $visitInformation)` make sure it is a dimension in scope `visit` then you can also remove the check for `in_array($dimension['scope'], array(self::SCOPE_ACTION, self::SCOPE_VISIT)) ` I think.